### DISCUSSION TITLE
sqlite and levenshtein distance

### DISCUSSION THREAD

**Discussion by mothsART** [diesel-rs/diesel#2989] [Q&A]
Hi,

I worked on a little cli soft who's a better **killall** command : https://github.com/mothsART/kouik
I would like to detect typing errors.
Best solution for this seems levenshtein distance.

Sqlite give an extension https://www.sqlite.org/spellfix1.html that I have compile :
file => https://github.com/mothsART/kouik/blob/master/spellfix.c
compile on debian with :
```
gcc -g -fPIC -shared spellfix.c -o spellfix.so
```

If I load manually this extension ".load ./spellfix", I can launch this SQL request :

```sql
SELECT * FROM programs WHERE editdist3(keyword, 'navugateur') < 200;
```

But I want to use it on diesel and I don't know how dynamicaly load extension and how I must edit my request to use it.

Currently :

```rust
let results = programs.filter(keyword.eq(program))
                      .filter(locale.eq(lang).or(locale.is_null()))
                      .load::<Program>(&connection);
```

Best regards

**Comment by weiznich** [comment]
Loading custom extensions is unfortunately not supported yet. There is https://github.com/diesel-rs/diesel/pull/2180, but thats not merged yet as there are open questions which need to be resolved.
